activesupport/lib/active_support/callbacks.rb

```diff
@@ -826,6 +754,30 @@ def get_callbacks(name)
       def set_callbacks(name, callbacks)
         send "_#{name}_callbacks=", callbacks
       end
+
+      def default_false_terminator
+        Proc.new do |target, result_lambda|
+          terminate = true
+          catch(:abort) do
+            result = result_lambda.call if result_lambda.is_a?(Proc)
+            if CallbackChain.halt_and_display_warning_on_return_false && result == false
```

Comment:
Since this terminator is getting moved *outside* of the `CallbackChain` class, we could simplify the code even more by replacing the current 

```ruby
ActiveSupport::Callbacks::CallbackChain.halt_and_display_warning_on_return_false
```

with 

```ruby
ActiveSupport::Callbacks.halt_and_display_warning_on_return_false
```
In other words, we could make `halt_and_display_warning_on_return_false` a module accessor.

Note that this change is independent of the rest, and independent of this commit.
I just think it would be nicer, but deserves its own commit, since we would also have to change any snippet of code where `halt_and_display_warning_on_return_false` (such as the CHANGELOG files)